EOLib/Domain/Character/GhostingRepository.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using AutomaticTypeMapper;
+
+namespace EOLib.Domain.Character
+{
+
+    public interface IGhostingRepository : IResettable
```

Comment:
This is mixing the concepts of `Repository` and `Service`

1. Services are stateless* classes that do a thing
2. Repositories or Providers are classes that store a thing (mapped to the same implementation in IoC, provider is read-only, repository is read/write)
3. Actions coordinate between services and repositories/providers. Data from repositories/providers is passed to services, operated on, and then updated in said repositories/providers accordingly.

So for this class: if you want a repository, break out the state into something with get/setters. Inject a provider if the consumer only needs read access, otherwise inject a repository. I think the operations are probably suited to `WalkValidationActions`, but the state should be stored in a provider/repository.

*Sometimes they aren't stateless, but the goal is that they should be. `InventoryService` is a good example of a service that doesn't hold any state and just does a thing. Same with `NumberEncoderService`.

Reply:
Resolved via local refactor (to be pushed)